### PR TITLE
Fix iceberg table can not join reorder

### PR DESCRIPTION
**Pull request by caneGuy**
## What type of PR is this：
- [x] enhancement

## Which issues of this PR fixes ：

related with #1030

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/external/iceberg/cost/IcebergTableStatisticCalculator.java [hunk 1]
```diff
@@ -35,6 +36,7 @@
 import static com.google.common.collect.ImmutableList.toImmutableList;
 import static java.util.stream.Collectors.toSet;
 
+// TODO @caneGuy add cache for statistics, currently getting stats from iceberg metafiles is not time consuming
```
Comment (dirtysalt) [inline fe/fe-core/src/main/java/com/starrocks/external/iceberg/cost/IcebergTableStatisticCalculator.java:39]: Yes, please support cache. I'm quite worried generating plan process will timeout once the scan file list is long. Plus, is there any possibility that same table stats has been computed for multiple times. For example, we have scan nodde on the same table but with different predicates.

Comment (caneGuy) [reply fe/fe-core/src/main/java/com/starrocks/external/iceberg/cost/IcebergTableStatisticCalculator.java:39]: how about add cache in an other pr? @dirtysalt

Comment (caneGuy) [reply fe/fe-core/src/main/java/com/starrocks/external/iceberg/cost/IcebergTableStatisticCalculator.java:39]: https://github.com/StarRocks/starrocks/issues/3497 i created a issue for this, and i will submit a pr next week

**Review by dirtysalt — DISMISSED** [review]

**Comment by caneGuy** [comment]
run starrocks_fe_unittest

**Review by Seaven — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/external/iceberg/cost/IcebergTableStatisticCalculator.java [hunk 2]
```diff
@@ -236,4 +200,103 @@ private void updatePartitionedStats(
...
+        }
+        return icebergFileStats;
+    }
+
+    private boolean generateColumnStatistic(IcebergFileStats icebergFileStats,
+                                            ColumnStatistic.Builder columnBuilder,
+                                            int fieldId,
+                                            double recordCount,
+                                            List<ColumnRefOperator> columnList) {
+        boolean isEstimated = true;
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/external/iceberg/cost/IcebergTableStatisticCalculator.java:265]: How about return ColumnStatistic instread of boolean, you  can return ColumnStatistic::UNKNOWN directly if isEstimated os false, and check isUnknown outside

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Utils.java [hunk 3]
```diff
@@ -376,6 +380,18 @@ public static boolean hasUnknownColumnsStats(OptExpression root) {
                     LOG.warn("hive table {} get column failed. error : {}", hiveTable.getName(), e);
                     return true;
                 }
+            } else if (operator instanceof LogicalIcebergScanOperator) {
+                IcebergTable table = (IcebergTable) scanOperator.getTable();
+                try {
+                    List<ColumnStatistic> columnStatisticList = IcebergTableStatisticCalculator.getColumnStatistics(
+                            // TODO: pass predicate to get column statistics
+                            new ArrayList<>(), table.getIcebergTable(),
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Utils.java:388]: why need to pass predicate？ It looks like pass colNames is enough

**Review by Youngwb — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Pull request merged**